R/paste_linter.R

```diff
@@ -111,6 +123,76 @@ paste_linter <- function(allow_empty_sep = FALSE, allow_to_string = FALSE) {
     /parent::expr
   "
 
+  slash_str <- sprintf("STR_CONST[%s]", xp_text_in_table(c("'/'", '"/"')))
+  str_not_start_with_slash <-
+    "STR_CONST[not(substring(text(), 2, 1) = '/')]"
+  str_not_end_with_slash <-
+    "STR_CONST[not(substring(text(), string-length(text()) - 1, 1) = '/')]"
+  non_str <- "SYMBOL or expr"
+
+  # Type I: paste(..., sep = "/")
+  paste_file_path_xpath <- glue("
+  //SYMBOL_FUNCTION_CALL[text() = 'paste']
```

Comment:
This could also catch `paste0(..., sep = "/")` - or is it already linted for any non-missing `sep=`?
The latter would be even better, although it would cause two "chained" lints for users refactoring `paste0("a", "b", sep = "/") = paste("a", "b", sep = "/") = file.path("a", "b")`